src/main/java/com/googlecode/download/maven/plugin/internal/HttpFileRequester.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright 2009-2018 The Apache Software Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.googlecode.download.maven.plugin.internal;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.ResponseHandler;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.protocol.HttpContext;
+import org.apache.maven.plugin.logging.Log;
+
+public class HttpFileRequester {
+
+    private static final long KBYTE = 1024L;
+    private static final char K_UNIT = 'K';
+    private static final char B_UNIT = 'b';
+
+    private final CloseableHttpClient httpClient;
+    private final Log log;
+
+    public HttpFileRequester(final CloseableHttpClient httpClient, final Log log) {
+        this.httpClient = httpClient;
+        this.log = log;
+    }
+
+    public void download(final URI uri, final File outputFile, final HttpContext clientContext) throws Exception {
+        final HttpGet httpGet = new HttpGet(uri);
+        httpClient.execute(httpGet, new ResponseHandler<Void>() {
```

Comment:
@ok2c not a fan of anonymous classes of such size. Let's extract it to named inner class